File: Farhaan/Week-2 Problems/maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int sum=0,gSum=INT_MIN,s=nums.size();
    
        for(int i=0;i<k;i++)
            sum+=nums[i];

        gSum=sum;

        for(int i=k;i<s;i++){
            sum-=nums[i-k];
            sum+=nums[i];

            gSum=max(gSum,sum);
        }
        return (gSum*1.0)/(k*1.0);
    }
};

File: Jhanavi/12-09-2022/Maximum_Average_Subarray_I.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
         window = max_sum = sum(nums[:k])
         for i in range(k, len(nums)):
            window += nums[i] - nums[i - k]
            max_sum = max(max_sum, window)
         return max_sum / k
